### PR TITLE
search results page

### PR DESCRIPTION
**Pull request by mirceabozi**

### CODE REVIEW

**Comment by spectro-prow** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/spectrocloud/librarium/pull/690#" title="Author self-approved">mirceabozi</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign 
You can assign the PR to them by writing `/assign ` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=spectrocloud%2Flibrarium).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/spectrocloud/librarium/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by spectro-prow** [comment]
Hi @mirceabozi. Thanks for your PR.

I'm waiting for a [spectrocloud](https://github.com/orgs/spectrocloud/people) member to verify that this patch is reasonable to test. If it is, they should reply with `/ok-to-test` on its own line. Until that is done, I will not automatically test new commits in this PR, but the usual testing commands by org members will still work. Regular contributors should [join the org](https://github.com/orgs/spectrocloud/people) to skip this step.

Once the patch is verified, the new status will be reflected by the `ok-to-test` label.

I understand the commands that are listed [here](https://go.k8s.io/bot-commands?repo=spectrocloud%2Flibrarium).

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

File: src/shared/components/SearchResults.js [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+  a {
+    font-size: 18px;
+    color: #206cd1;
+  }
+`;
+
+export const Hit = ({ hit }) => (
+  <HitWrap>
+    <div>
```
Comment (IonitaCatalina) [inline src/shared/components/SearchResults.js:51]: is this needed?

File: src/shared/components/SearchResults.js [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+  Index,
+  InstantSearch,
+  SearchBox,
+  Highlight,
+  Snippet,
+} from "react-instantsearch-dom";
+import algoliasearch from "algoliasearch/lite";
+import config from "../../../config";
+
+const HiddenSearchBox = styled(SearchBox)`
```
Comment (IonitaCatalina) [inline src/shared/components/SearchResults.js:18]: is this needed?

File: src/shared/components/SearchResults.js [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+  const searchClient = algoliasearch(
+    config.header.search.algoliaAppId,
+    config.header.search.algoliaSearchKey
+  );
+  return (
+    <InstantSearch
+      searchState={{ query: term }}
+      searchClient={searchClient}
+      indexName={`${config.header.search.indexName}`}
```
Comment (IonitaCatalina) [inline src/shared/components/SearchResults.js:76]: no need for templates

**Review by IonitaCatalina — APPROVED** [review]

**Comment by spectro-prow** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/spectrocloud/librarium/pull/690#pullrequestreview-960353434" title="Approved">IonitaCatalina</a>*, *<a href="https://github.com/spectrocloud/librarium/pull/690#" title="Author self-approved">mirceabozi</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign 
You can assign the PR to them by writing `/assign ` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=spectrocloud%2Flibrarium).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/spectrocloud/librarium/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: src/shared/components/SearchResults.js [hunk 4]
```diff
@@ -55,8 +56,23 @@ const Hit = ({ hit }) => (
 
 const HiddenSearchBox = connectSearchBox(() => null);
 
+const Results = connectStateResults(({ term, searchResults: res }) => {
+  if (!term) {
+    return <h1>Please insert a keyword in the search bar</h1>;
+  }
+  if (res?.nbHits === undefined) {
```
Comment (codrin-iftimie) [inline src/shared/components/SearchResults.js:63]: `typeof res?.nbHits === "undefined"`

**Review by codrin-iftimie — CHANGES_REQUESTED** [review]

**Comment by spectro-prow** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/spectrocloud/librarium/pull/690#issuecomment-1120741156" title="Approved">codrin-iftimie</a>*, *<a href="https://github.com/spectrocloud/librarium/pull/690#pullrequestreview-960353434" title="Approved">IonitaCatalina</a>*, *<a href="https://github.com/spectrocloud/librarium/pull/690#" title="Author self-approved">mirceabozi</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign 
You can assign the PR to them by writing `/assign ` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=spectrocloud%2Flibrarium).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/spectrocloud/librarium/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**